Add dev branch

Taras2023 opened the pull request.

Commit:
Add dev branch

Taras2023 committed:
Проектная работа 1

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,60 @@
+    Calculator(){
+        numberPeople();
+        addProduct();
+        totals();
```
Не очень правильно делать основную работу класса в конструкторе, он нужен для инициализации, передачи аргументов снаружи

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,60 @@
+
+        while (true) {
+            System.out.println("На скольки человек необходимо разделить счет?");
+            number = scanner.nextInt();
```
⚠️ Если передать в этот момент произвольную строку программа завершит работу с исключением. Перед тем, как прочитать инт из сканнера, нужно сначала проверить возможно ли это при помощи соответствующего метода `hasNextInt`

gusar-off commented on src/main/java/AddProduct.java:
```diff
@@ -0,0 +1,19 @@
+        System.out.println("Введите название товара");
+        this.product = scanner.next();
+        System.out.println("Введите цену товара");
+        this.price = scanner.nextFloat();
```
⚠️ Тут тоже самое, только `hasNextFloat `

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,60 @@
+        else if (lastnumber==1)
+            rub = "рубль";
+        else
+            rub = "рубля";
```
⚠️Немного некорректно рассчитывается окончание. Получается если цена будет 112 рублей, метод вернет рубля. Правильный алгоритм звучит так - Проверить лежит ли остаток от деления на 100 в интервале от 11 до 14 включительно, если да, то возвращаем рублей, если нет, то берем остаток от деления на 10 и проверяем - если 1, то рубль, 2-4 то рубля, в остальных случаях - рублей. То есть нужно во-первых проверять именно остаток от деления, во-вторых учесть, что числа заканчивающиеся на 11-12-13-14 и заканчивающиеся на 1-2-3-4, будут иметь разные окончания

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,60 @@
+            AddProduct nomenclature = new AddProduct();
+            amount += nomenclature.price;
+            products = products +"\n"+ nomenclature.product;
+            System.out.println("Вы хотите добавить еще один товар?");
```
С точки зрения пользователя было бы удобнее здесь уточнить, что нужно ввести в консоль, чтобы завершить добавление товаров

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,60 @@
+    public void totals(){
+
+        System.out.println("Добавленные товары: \n" + products);
+        System.out.println("Стоимость товаров: " + amount);//String.format("%.2f",amount));
```
Закомментированный код лучше убрать.

Taras2023 committed:
1. Убрал конструкторы
2. Исправил ввод цены
3. Исправил ввод количества человек
4. Исправил окончание слова рубль

gusar-off commented on src/main/java/AddProduct.java:
```diff
@@ -5,15 +5,28 @@ public class AddProduct {
 
+            } else {
+                System.out.println("Введите положительное число");
+                scanner = new Scanner(System.in);
```
Тут и выше можно не создавать каждый раз новый инстанс класса `Scanner `
